apis/python/src/tiledb/vector_search/object_readers/directory_reader.py

```diff
@@ -8,6 +18,223 @@
 # from tiledb.vector_search.object_readers import ObjectPartition, ObjectReader
 
 
+# Util functions for file matching and recursive listing
+def match_uri(
```

Comment:
(Minor)(Optional) Could add unit tests for `match_uri()` / `find_uris_vfs()` / `find_uris_tiledb_group()` etc.

Reply:
Added tests for this